store/v2/commitment/metadata.go

```diff
@@ -99,6 +100,53 @@ func (m *MetadataStore) flushCommitInfo(version uint64, cInfo *proof.CommitInfo)
 	return batch.Close()
 }
 
+func (m *MetadataStore) flushRemovedStoreKeys(version uint64, storeKeys []string) (err error) {
+	batch := m.kv.NewBatch()
+	defer func() {
+		err = batch.Close()
+	}()
+
+	for _, storeKey := range storeKeys {
+		key := []byte(fmt.Sprintf("%s%s", encoding.BuildPrefixWithVersion(removedStoreKeyPrefix, version), storeKey))
+		if err := batch.Set(key, []byte{}); err != nil {
+			return err
+		}
+	}
+	return batch.WriteSync()
+}
+
+func (m *MetadataStore) deleteRemovedStoreKeys(version uint64, fn func(storeKey []byte) error) (err error) {
```

Comment:
nit: is `fn` a filter function? if so can it have a more descriptive name? kind of hard to read